anesthetic/plot.py

```diff
@@ -402,6 +472,77 @@ def contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
     return cbar
 
 
+def kde_contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
+    """Plot a 2d marginalised distribution as contours.
+
+    This functions as a wrapper around matplotlib.axes.Axes.tricontour, and
+    matplotlib.axes.Axes.tricontourf with a kernel density estimation
+    computation provided by scipy.stats.gaussian_kde in between. All remaining
+    keyword arguments are passed onwards to both functions.
+
+    Parameters
+    ----------
+    ax: matplotlib.axes.Axes
+        axis object to plot on.
+
+    data_x, data_y: numpy.array
+        x and y coordinates of uniformly weighted samples to generate kernel
+        density estimator.
+
+    weights: numpy.array, optional
+        Sample weights.
+
+    ncompress: int, optional
+        Degree of compression. Default 1000
+
+    xmin, xmax, ymin, ymax: float
+        lower/upper prior bounds in x/y coordinates.
+        optional, default None
+
+    Returns
+    -------
+    c: matplotlib.contour.QuadContourSet
+        A set of contourlines or filled regions
+
+    """
+    xmin = kwargs.pop('xmin', None)
+    xmax = kwargs.pop('xmax', None)
+    ymin = kwargs.pop('ymin', None)
+    ymax = kwargs.pop('ymax', None)
+    weights = kwargs.pop('weights', None)
+    n = kwargs.pop('ncompress', 1000)
```

Comment:
`n` is used just once in this function, so I don't see the need for the abbreviation from `ncompress`. I'd stick with the more descriptive name (also for 1D case).